### PR TITLE
linux: pci: Fix PCI probing

### PR DESCRIPTION
**Pull request by diogo-ivo**
The Keystone PCI controller had a race condition hidden that made its
probing fail. Add a patch provided by TI (currently mainlined) that
fixes this race condition and returns PCI back to a funcional state.
Tested with a Hailo-8 NPU card.

While at it add a second patch suggested by TI that fixes "a 45 second
delay when no Endpoint Device is connected to the PCIe connector".

### CODE REVIEW

**Comment by jan-kiszka** [comment]
And do they fix the issues we've seen? This remains unclear to me when just reading this PR.

If so, it would we good to also request them to go into stable.

**Comment by diogo-ivo** [comment]
Yes, this fixes the probing and now the Hailo device is seen when running lspci

**Comment by diogo-ivo** [comment]
Regarding pushing them to stable I'll send an e-mail tomorrow

File: recipes-kernel/linux/files/patches-6.1/0086-PCI-keystone-Fix-race-condition-when-initializing-PH.patch [hunk 1]
```diff
@@ -0,0 +1,61 @@
+From 36836820f571872d6856c17bdc1d0e4f8418915d Mon Sep 17 00:00:00 2001
```
Comment (BaochengSu) [inline recipes-kernel/linux/files/patches-6.1/0086-PCI-keystone-Fix-race-condition-when-initializing-PH.patch:1]: The patch file should be generated from:

```shell
git format-patch --abbrev=12 --no-numbered --zero-commit --no-signature ...
```

see: https://github.com/siemens/meta-iot2050/blob/master/CONTRIBUTING.md#patch-files-generation

Meanwhile I will test the SSD on M.2 variant.

Comment (diogo-ivo) [reply recipes-kernel/linux/files/patches-6.1/0086-PCI-keystone-Fix-race-condition-when-initializing-PH.patch:1]: Ok, I'll prepare the patches like that. Please let me know the results of your testing when you can so I can reply to TI

Comment (BaochengSu) [reply recipes-kernel/linux/files/patches-6.1/0086-PCI-keystone-Fix-race-condition-when-initializing-PH.patch:1]: Now with the patch my PM971 256G SSD is working.

**Review by BaochengSu — CHANGES_REQUESTED** [review]
Please rebase the patches - change the number, sorry that #520 merged firstly 😉

**Comment by BaochengSu** [comment]
> Please rebase the patches - change the number, sorry that #520 merged firstly 😉

Never mind, I plan to merge the #511 first, then I will do the rebase for this PR.

**Review by BaochengSu — APPROVED** [review]

**Pull request merged**